### ISSUE TITLE
project.properties - allow include/exclude options

### ISSUE THREAD

**Issue by hramadan** [particle-iot/particle-cli#410]
I understand that .particle.ignore and .particle.include are no longer used by the compiler and project.properties is used for configuration. 

The ability to include/exclude files including sub-directories would provide a "make" like facility to selectively compile sets of source code files to build different binaries (say product A and product B) from a single repository. 

This would make the build process straight forward - to build product A rename or link project.properties.productA to project.properties. 

In short, can .particle.ignore and .particle.include facilities be included into project.properties.

**Comment by monkbroc** [comment]
I'm hesitant to add more complex mechanics to the compile command. I would recommend using a tool like `make` for dynamically generating the list of files that fit your needs exactly.

I still like your proposal because it is simple.
